### PR TITLE
travis: update CI environments

### PR DESCRIPTION
**Pull request by dannys42**
- Remove Swift 4.0.x support on macOS
- Update Travis to use Kitura CI docker images

### CODE REVIEW

File: Sources/BlueSocketTestCommonLibrary/ServerOperation.swift [hunk 1]
```diff
@@ -32,24 +32,21 @@ public class ServerOperation: Operation {
...
         }
 
         while !self.isCancelled {
-            autoreleasepool {
-                let socketHandlers: [SocketHandler] = [serverHandler] + clientHandlers
-                
-                let activeHandlers = socketHandlers.wait(timeout: 12)
-                activeHandlers.process()
-            }
+            let socketHandlers: [SocketHandler] = [serverHandler] + clientHandlers
```
Comment (mbarnach) [inline Sources/BlueSocketTestCommonLibrary/ServerOperation.swift:46]: @dannys42 What is the reason to remove the `autoreleasepool`?

Comment (dannys42) [reply Sources/BlueSocketTestCommonLibrary/ServerOperation.swift:46]: It looks like autoreleasepool is not available on Linux.  So I'm not quite sure how to handle this situation... perhaps we need to do some sort of `#if macOS` type of thing here?

**Review by mbarnach — APPROVED** [review]
👍

**Review by mbarnach — COMMENTED** [review]
Nice one!

**Comment by dannys42** [comment]
@mbarnach Thanks!  Sorry it took a while, but it's now working and integrated into TravisCI.com

**Comment by mbarnach** [comment]
Awesome!

**Pull request merged**